core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -444,12 +444,15 @@ private[spark] object JsonProtocol {
     ("Disk Size" -> rddInfo.diskSize)
   }
 
-  def storageLevelToJson(storageLevel: StorageLevel): JValue = {
-    ("Use Disk" -> storageLevel.useDisk) ~
-    ("Use Memory" -> storageLevel.useMemory) ~
-    ("Deserialized" -> storageLevel.deserialized) ~
-    ("Replication" -> storageLevel.replication)
-  }
+  def storageLevelToJson(storageLevel: StorageLevel): JValue =
+    storageLevel.name match {
```

Comment:
nit:
```
storageLevel.name.getOrElse(...)
```

Reply:
Done. Sadly, in this case, getOrElse requires explicit type.

Reply:
Actually, after seeing the compilation error, I recall why I went with a match instead of `getOrElse` -- the former does not require an explicit conversion to JString.